### PR TITLE
feat: improve UI and responsiveness of file selector

### LINKED ISSUE #1011: Improve file selector UI and responsiveness
### Tasks

 - [ ] Show close icon on the top right side as usual
 - [ ] Correctly render the close button as component and icon (currently manually written as button and hardcoded with letter 'x')
 - [ ] Make it responsive
 - [ ] Add (optional) title to the file selector modal
 - [ ] Improve whole layout of the modal

file-selector_bad

file-selector_responsive-bad

### Use case

_No response_

### Related Pull request

_No response_

### PR DESCRIPTION
**Pull request by tschawax**
Render close icon as component and at the top right.
Add new optional heading to modal component.
Make file selector layout responsive.
Add various small design improvements.

Closes #1011

before:
file-selector_bad(1)

after:
file-selector-improved(1)

before:
277717164-2a5f37be-9818-4a97-941d-3997418e3604

after:
file-selector-improved(2)

## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

**Review by markus2330 — APPROVED** [review]
Great job, we love the changes! :heart: 

Are all of these changes already described in the UI guidelines?
I think reworking such code is always a good opportunity to check if UI guidelines are complete.

@absurd-turtle do you want check these changes here and the UI guidelines?

File: frontend/src/features/nextcloud_integration/components/FileSelectorModal.tsx [hunk 1]
```diff
@@ -35,17 +40,16 @@ export default function FileSelectorModal({
...
-            {/* <h2>{title}</h2> */}
-            <button
-              onClick={onCancel}
-              className="rounded-lg border border-neutral-600 px-5 py-2.5 text-center text-sm font-medium hover:bg-neutral-600 focus:outline-none focus:ring-4 focus:ring-blue-300 sm:w-auto"
-            >
-              x
-            </button>
-          </div>
-          <div className="flex h-full max-h-[60vh] w-full justify-center p-4">
+        <div className="flex h-[50vh] w-[90vw] flex-col rounded-lg bg-neutral-100 p-1 dark:bg-neutral-100-dark sm:w-[80vw] md:w-[70vw] md:p-6 lg:w-[60vw] xl:w-[50vw] 2xl:w-[40vw]">
```
Comment (absurd-turtle) [inline frontend/src/features/nextcloud_integration/components/FileSelectorModal.tsx:43]: image
Consider making the height bigger again or responsive to viewports with smaller heights.
It gets pretty unusable when the window is half the height of my monitor.

**Review by absurd-turtle — COMMENTED** [review]
I like the changes! Very nice :)
However the layout still has some problems with responsiveness:

For me text overlaps at small viewports.
Maybe change the overlap css attribute for the items.
Problem with that is that the full filename can not be read which might be a problem as well.
image

**Comment by absurd-turtle** [comment]
> I like the changes! Very nice :) However the layout still has some problems with responsiveness:
> 
> For me text overlaps at small viewports. Maybe change the overlap css attribute for the items. Problem with that is that the full filename can not be read which might be a problem as well. image

Tbh I don't know if we necessarily need the last modified meta data but it is nice for sorting :D

**Comment by markus2330** [comment]
I merged as this seems to be a problem only for mobile devices.

**Pull request merged**